files/default/neutron-ha-tool.py

```diff
@@ -585,8 +589,58 @@ def migrate_router(qclient, router_id, agent, target):
     if router_id not in list_routers_on_l3_agent(qclient, target['id']):
         raise RuntimeError("Failed to add router_id=%s from agent_id=%s" %
                            (router_id, agent['id']))
+    if wait_for_ports:
+        wait_router_ports_active(qclient, router_id, target['host'])
 
 
+def wait_router_ports_active(qclient, router_id, target_host, maxtries=60):
+    """
+    Returns nothing. Waits for all non-distributed ports of a router
+    for being in the ACTIVE status on a specific host
+
+    :param qclient: A neutron client
+    :param router_id: The id of the router to check
+    :param target_host: The host on which the ports should be ACTIVE
+                        (reflected by the binding:host_id attribute of
+                        a port)
+    :param maxtries: Maximum number of times the ports' status is polled,
+                     an expection is raised there are still ports with a
+                     non ACTIVE status after that.
+    """
+
+    active_port_ids = set()
+    remaining = ['dummy']
+    LOG.info("Wait for the ports of router_id=%s to be ACTIVE" % router_id)
+    while (maxtries > 0 and len(remaining)):
```

Comment:
`while remaining and maxtries:` can do the trick, also I was thinking how to populate `remaining` before, and move the `maxtries` check inside the while and ends it with a `break`